### PR TITLE
Refactor ingest workflow to use current working file

### PR DESCRIPTION
**Pull request by awead**
Refs projecthydra/sufia#2553

Changes proposed in this pull request:

* FileActor only downloads a working file unless the uploaded file is unavailable locally
* IngestFileJob now accepts optional parameters, such as filename, which had been causing unnecessary downloads because the tempfile had a different name than the original.
* The current working file is passed from IngestFileJob to successive jobs like CharactizeJob and CreateDerivativeJob

This avoids the problem of downstream applications forcing another download of the file because the working file may be in a different, yet local, location. In this case, once the job has the local file "in hand," it passes it to successive jobs as an optional argument. If the job receives the local file argument, it users CurationConcerns::WorkingDirectory to check if the file still exists and re-downloads it necessary. 

@projecthydra/sufia-code-reviewers

### CODE REVIEW

**Comment by awead** [comment]
Comments welcome. I'm looking at how this will solve the referenced issue in Sufia.

File: app/services/curation_concerns/working_directory.rb [hunk 1]
```diff
@@ -1,10 +1,12 @@
 module CurationConcerns
-  module WorkingDirectory
+  class WorkingDirectory
```
Comment (jcoyne) [inline app/services/curation_concerns/working_directory.rb]: This class is never instantiated, was this change necessary?

File: app/jobs/ingest_file_job.rb [hunk 2]
```diff
@@ -2,36 +2,42 @@ class IngestFileJob < ActiveJob::Base
...
                                         local_file,
-                                        relation.to_sym,
+                                        relation,
                                         versioning: false)
 
     # Persist changes to the file_set
     file_set.save!
 
-    repository_file = file_set.send(relation.to_sym)
+    repository_file = file_set.send(relation)
```
Comment (jeremyf) [inline app/jobs/ingest_file_job.rb]: Are we calling a public method? If so, prefer `#public_send` over `#send`.

File: app/jobs/ingest_file_job.rb [hunk 3]
```diff
@@ -2,36 +2,42 @@ class IngestFileJob < ActiveJob::Base
...
-  # @param [String] relation ('original_file')
-  def perform(file_set, filename, mime_type, user, relation = 'original_file')
-    local_file = File.open(filename, "rb")
+  # @option opts [String] mime_type
+  # @option opts [String] filename
+  # @option opts [String] relation, ex. :original_file
+  def perform(file_set, filepath, user, opts = {})
+    mime_type = opts.fetch(:mime_type, nil)
+    filename = opts.fetch(:filename, File.basename(filepath))
+    relation = opts.fetch(:relation, :original_file).to_sym
```
Comment (jeremyf) [inline app/jobs/ingest_file_job.rb]: Is `#to_sym` required? Would `#to_s` suffice? Creating symbols prior to Ruby 2.2 (or so) means creating objects that are not cleaned up.

Comment (awead) [inline app/jobs/ingest_file_job.rb]: @jeremyf yeah, because ActiveJob doesn't allow symbols as parameters unfortunately.

File: app/services/curation_concerns/working_directory.rb (hunk 1 above)
Comment (awead) [inline app/services/curation_concerns/working_directory.rb]: In case anyone wants to extend it. It was leftover from an earlier attempt with Sufia integration when I was trying to create:
`Sufia::WorkingDirectory < CurationConcerns::WorkingDirectory` I'm ambivalent about it now.

File: app/jobs/ingest_file_job.rb (hunk 3 above)
Comment (jcoyne) [inline app/jobs/ingest_file_job.rb]: Prior to this PR it was already doing `.to_sym`, so let's not address this now.

File: app/jobs/ingest_file_job.rb (hunk 2 above)
Comment (jcoyne) [inline app/jobs/ingest_file_job.rb]: Good suggestion, but not required in this PR as this is the way it was before.
